### PR TITLE
Update assertk

### PR DESCRIPTION
**Pull request by eygraber**

### CODE REVIEW

File: kotlin-inject-compiler/test/src/test/kotlin/me/tatarka/inject/test/Utils.kt [hunk 1]
```diff
@@ -11,4 +14,10 @@ import javax.tools.Diagnostic
...
 fun Assert<TestCompilationResult>.warnings(): Assert<String> =
-    prop("warnings") { it.output(Diagnostic.Kind.WARNING) }
\ No newline at end of file
+    prop("warnings") { it.output(Diagnostic.Kind.WARNING) }
+
+internal inline fun <T> assertSuccess(block: () -> T) =
+    assertThat(runCatching(block)).isSuccess()
+
+internal inline fun <T> assertFailure(block: () -> T) =
+    assertThat(runCatching(block)).isFailure()
```
Comment (evant) [inline kotlin-inject-compiler/test/src/test/kotlin/me/tatarka/inject/test/Utils.kt:23]: You shouldn't need to add wrappers for these. `assertFailue` should exist in assertk and
```kotlin
assertThat { ... }.isSuccess()
```
can be replaced with just
```kotlin
assertThat(...)
```

Comment (eygraber) [reply kotlin-inject-compiler/test/src/test/kotlin/me/tatarka/inject/test/Utils.kt:23]: I think that was deprecated

Comment (eygraber) [reply kotlin-inject-compiler/test/src/test/kotlin/me/tatarka/inject/test/Utils.kt:23]: Oh nevermind, I see what you're saying

File: kotlin-inject-compiler/test/src/test/kotlin/me/tatarka/inject/test/RoundsTest.kt [hunk 2]
```diff
@@ -41,13 +40,13 @@ class RoundsTest {
                     }
                 """.trimIndent()
             ).compile()
-        }.isSuccess()
+        )
```
Comment (evant) [inline kotlin-inject-compiler/test/src/test/kotlin/me/tatarka/inject/test/RoundsTest.kt:43]: another assertThat without an assertion

Comment (evant) [reply kotlin-inject-compiler/test/src/test/kotlin/me/tatarka/inject/test/RoundsTest.kt:43]: oh nvm it's the same one

**Review by evant — APPROVED** [review]

**Pull request merged**